server/etcdserver/server.go

```diff
@@ -1768,6 +1772,12 @@ func (s *EtcdServer) LeaderChangedNotify() <-chan struct{} {
 	return s.leaderChanged
 }
 
+func (s *EtcdServer) FirstCommitInTermNotify() <-chan struct{} {
```

Comment:
Please comment the semantic.

Reply:
I was influenced by 
https://github.com/etcd-io/etcd/blob/06a627abd4a717b63be7162a766d7cd882fd56a4/server/etcdserver/server.go#L1769
but now I don't see any good reason to make `FirstCommitInTermNotify` exported. I'll change it to `firstCommitInTermNotify`. 
Do you think that it still should be commented once it is not exported?

Reply:
I think it might be public. It seems to be equally good (if not better) mechanism to react in embedded server to leadership changes. 

Either way: it's good to comment even private methods / fields.  If package has 10k lines, everything there is like 'public'.

Reply:
Do you think, that it should be part of 
https://github.com/etcd-io/etcd/blob/06a627abd4a717b63be7162a766d7cd882fd56a4/server/etcdserver/server.go#L155
then (as `LeaderChangedNotify` is)?

Reply:
I would ask for the first user that wants the method to be exposed and then expose it, vs exposing it proactively. 

BTW:  Please add an integration test whether the notification really happens during leadership change.